inspectit-oce-core/src/test/java/rocks/inspectit/oce/core/instrumentation/context/InspectitContextUnitTest.java

```diff
@@ -0,0 +1,748 @@
+                assertThat(getCurrentTagsAsMap()).containsEntry("myTag", "myValue");
+                assertThat(getCurrentTagsAsMap()).containsEntry("rootKey", "rootValue");
+
+                ctxA.close();
+            }
+
+            root.close();
+            assertThat(InspectitContext.INSPECTIT_KEY.get()).isNull();
+        }
+
+
+        @Test
+        void verifyDataOnlyPublishedAsTagWhenConfigured() {
+            when(commonTags.getCommonTagValueMap()).thenReturn(Collections.emptyMap());
+            doReturn(true).when(propagation).isPropagatedDownWithinJVM(any());
+            doReturn(true).when(propagation).isTag(eq("my_tag"));
+
+            InspectitContext root = InspectitContext.createFromCurrent(commonTags, propagation, true);
+            root.setData("my_tag", "tagValue");
+            root.setData("my_hidden", "hiddenValue");
+
+            root.makeActive(true);
+
+            assertThat(getCurrentTagsAsMap()).hasSize(1);
+            assertThat(getCurrentTagsAsMap()).containsEntry("my_tag", "tagValue");
+
+            root.close();
+            assertThat(InspectitContext.INSPECTIT_KEY.get()).isNull();
+        }
+
+
+        @Test
+        void verifyDataTypesPreservedWithinTrace() {
+            when(commonTags.getCommonTagValueMap()).thenReturn(Collections.emptyMap());
+            doReturn(true).when(propagation).isPropagatedDownWithinJVM(any());
+            doReturn(true).when(propagation).isTag(any());
+
+            InspectitContext root = InspectitContext.createFromCurrent(commonTags, propagation, true);
+            root.setData("rootKey", "rootValue");
+            root.setData("myTag", "rootValue");
+            root.setData("longKey", 42L);
+
+            root.makeActive(true);
+
+            TagContextBuilder tcb = Tags.getTagger().currentBuilder()
+                    .put(TagKey.create("myTag"), TagValue.create("myValue"));
+            try (Scope tc = tcb.buildScoped()) {
+
+                Map<String, String> currentTagsAsMap = getCurrentTagsAsMap();
+                assertThat(currentTagsAsMap).containsEntry("longKey", "42");
+
+                InspectitContext ctxA = InspectitContext.createFromCurrent(commonTags, propagation, true);
+                ctxA.makeActive(true);
+                assertThat(ctxA.getData("myTag")).isEqualTo("myValue");
+                assertThat(ctxA.getData("rootKey")).isEqualTo("rootValue");
+                assertThat(ctxA.getData("longKey")).isEqualTo(42L);
+
+                ctxA.close();
+            }
+
+            root.close();
+            assertThat(InspectitContext.INSPECTIT_KEY.get()).isNull();
+        }
+
+
+        @Test
+        void verifyCommonTagsPublished() {
+            HashMap<String, String> tags = new HashMap<>();
+            tags.put("tagA", "valueA");
+            tags.put("tagB", "valueB");
+            when(commonTags.getCommonTagValueMap()).thenReturn(tags);
+            doReturn(true).when(propagation).isPropagatedDownWithinJVM(any());
```

Comment:
shouldn't the mocking of the propagation be removed in this test? or actually return `false` for both

Reply:
The mocking just ensures the default behaviour of the data configuration. This is needed for makeActive to know that it should publish its data, which includes the common tags as a tag scope.